core/inc/knoting/engine.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <array>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "Window.h"
```

Comment:
`window.h` instead of `Window.h`.  
And do we want to make the convention that we decide upon follow inside the headers, `<knoting/window.h>` instead of `"window.h"`, or keep it this way?